packages/core/utils/prompts.ts

```diff
@@ -1,10 +1,17 @@
 import { cancel, intro, isCancel, outro, select, text, multiselect } from "@clack/prompts";
 
-export type PromptOption = {
-    label: string;
-    value: boolean | string;
-    hint?: string;
-};
+type Primitive = Readonly<string | boolean | number>;
+export type PromptOption<Value> = Value extends Primitive
+    ? {
+          value: Value;
+          label?: string;
+          hint?: string;
+      }
+    : {
+          value: Value;
+          label: string;
+          hint?: string;
+      };
```

Comment:
Am I not seeing correctly, or is the only difference between the two types the optional label? Shouldn't the first type already include the second type?

Why are you declaring it explicitly?

Reply:
This type was sourced directly from `@clacks/prompts` as they don't expose it (for some reason 🤷): https://github.com/natemoo-re/clack/blob/45ee73bf33b25f9a8c7e1bb6117ccc165478bf4d/packages/prompts/src/index.ts#L208-L212

Without it, the types four our prompt wrappers aren't satisfied